podpac/core/node.py

```diff
@@ -485,266 +525,160 @@ def del_cache(self, key=None, coordinates=None):
-        except:
-            import _pickle as cPickle
-
-        try:
-            import boto3
-        except:
-            boto3 = None
-
-        path = self.cache_path(filename)
-        if settings.S3_BUCKET_NAME is None or not settings.CACHE_TO_S3:
-            with open(path, 'rb') as fid:
-                obj = cPickle.load(fid)
-        else:
-            s3 = boto3.resource('s3').Bucket(settings.S3_BUCKET_NAME)
-            io = BytesIO()
-            s3.download_fileobj(path, io)
-            io.seek(0)
-            obj = cPickle.loads(io.read())
-        return obj
+        if self.debug:
+            self._output = data
 
-    def clear_disk_cache(self, attr='*', node_cache=False, all_cache=False):
-        """Helper function to clear disk cache.
+        return data
+    return wrapper
 
-        WARNING: This function will permanently delete cached values
+def cached_property(key, depends=None, raise_no_cache_exception=False):
+    """
+    Decorating for caching a function's output based on a key. 
 
-        Parameters
-        ----------
-        attr : str, optional
-            Default '*'. Specific attribute to be cleared for specific
-            instance of this Node. By default all attributes are cleared.
-        node_cache : bool, optional
-            Default False. If True, will ignore `attr` and clear all attributes
-            for all variants/instances of this Node.
-        all_cache : bool, optional
-            Default False. If True, will clear the entire podpac cache.
-
-        .. deprecated:: 0.2.0
-            This method will be removed and replaced by the caching module by version 0.2.0.
-        """
+    Parameters
+    -----------
+    key: str
+        Key used for caching.
+    depends: str, list, traitlets.All (optional)
+        Default is None. Any traits that the cached property depends on. The cached_property may NOT
+        change the value of any of these dependencies (this will result in a RecursionError)
+    raise_no_cache_exception: bool, optional
+            Raises a NodeException if trying to put data to the cache, but no cache is available.
 
-        import warnings
-        warnings.warn('Node.clear_disk_cache will be replaced by del_cache in a later release', DeprecationWarning)
 
-        import glob
-        import shutil
+    Notes
+    -----
+    This decorator cannot handle function input parameters. 
+    
+    If the function uses any tagged attributes, these will essentially operate like dependencies
+    because the cache key changes based on the node definition, which is affected by tagged attributes.
 
-        if all_cache:
-            shutil.rmtree(settings.CACHE_DIR)
-        elif node_cache:
-            shutil.rmtree(self.cache_dir)
-        else:
-            for f in glob.glob(self.cache_path(attr)):
-                os.remove(f)
+    Examples
+    ----------
+    >>> class MyClass(Node):
+           value = 0
+           @cached_func('native_coordinates')
```

Comment:
Example should be consistent with the name of the decorator (e.g. `@cached_property('native_coordinates')`). 

But maybe it should be  `@cached_func` so as not to be confused with python `@property` which allows for the sytanx of accessing a function like it was a attribute (e.g. without the parenthesis.

Reply:
I meant to flag the fact that I wasn't totally happy with the name. I think `cache_func` is better. I will rename.